### PR TITLE
Add merges to CheckAndProcessElderChange

### PR DESCRIPTION
**Pull request by octol**
Add merges to CheckAndProcessElderChange flow tests

Render of graphs used: 
~https://raw.githack.com/maidsafe/routing/bd2d7d32cb1fcffc57e652df3fa5ff0f92f430fe/routing_model/index.html~
https://raw.githack.com/maidsafe/routing_model/3fce6d4e0ebce93ba027126476e9158e3f0393e9/routing_model/index.html

The sub-process ProcessMerge is missing and would be the next logical step

### CODE REVIEW

File: src/actions.rs [hunk 1]
```diff
@@ -35,6 +35,7 @@ pub struct InnerAction {
     pub section_members: BTreeMap<SectionInfo, Vec<Node>>,
 
     pub merge_infos: Option<MergeInfo>,
+    pub merge_needed: bool,
```
Comment (jeanphilippeD) [inline src/actions.rs]: Could you add one test case that set it to true (on top of the existing ones that have false).?

Comment (octol) [reply src/actions.rs]: Given that it's currently mocked, wouldn't such a test set the bool and then assert that it's actually set?

Comment (octol) [reply src/actions.rs]: I mean, currently since it's mocked there is no path in the flows that triggers it

**Comment by pierrechevalier83** [comment]
Note: This PR was moved from routing. Initial review comments here: https://github.com/maidsafe/routing/pull/1600

File: src/scenario_tests.rs [hunk 2]
```diff
@@ -183,6 +184,7 @@ fn run_test(
             action_our_events: action.our_events,
             action_our_section: action.our_section,
             action_merge_infos: action.merge_infos,
+            action_merge_needed: action.merge_needed,
```
Comment (jeanphilippeD) [inline src/scenario_tests.rs]: This should not be here, we set it ourselves so we don't want to check its value.

Comment (octol) [reply src/scenario_tests.rs]: Good catch! I'll remove it and squash

**Pull request merged**